lib/azure/armrest/storage_account_service.rb

```diff
@@ -108,33 +107,81 @@ def create(account_name, rgroup = configuration.resource_group, options)
         acct
       end
 
-      # Returns the primary and secondary access keys for the given
-      # storage account as a hash.
+      # Returns the primary and secondary access keys for the given storage
+      # account. This method will return a hash with 'key1' and 'key2' as its
+      # keys.
+      #
+      # If you want a list of StorageAccountKey objects, then use the
+      # list_account_keys2 method instead.
       #
       def list_account_keys(account_name, group = configuration.resource_group)
         validate_resource_group(group)
 
         url = build_url(group, account_name, 'listKeys')
         response = rest_post(url)
-        JSON.parse(response)
+        hash = JSON.parse(response.body)
+
+        parse_account_keys_from_hash(hash)
+      end
+
+      # Returns a list of StorageAccountKey objects consisting of information
+      # the primary and secondary keys. This method requires an api-version
+      # string of 2016-01-01 or later, or an error is raised.
+      #
+      # If you want a plain hash, use the list_account_keys method instead.
+      #
+      def list_account_keys2(account_name, group = configuration.resource_group)
```

Comment:
@djberg96 This is just a nit, but `list_account_key_objects` might be a better name for this method.